src/pymmcore_widgets/_mda/_positions_table_widget.py

```diff
@@ -384,7 +384,7 @@ def _add_table_row(
         row: int | None = None,
     ) -> None:
         if not self._mmc.getXYStageDevice() and not self._mmc.getFocusDevice():
-            raise ValueError("No XY and Z Stage devices selected.")
+            return
```

Comment:
i find this behavior (do nothing and warn nothing) a little jarring.  Why not just add a position at (0,0,0) rather than aborting altogether?

Reply:
That's because since no xy and z stage are loaded, there are no columns visible in the position widget. So even If we add (0,0,0) there won't be anything in the table. If we want to allow this, I think  we need to completely remove the hide/show behavior of the widget. Doe you think it make sense?

Reply:
And why do we hide stuff?  When is it useful?

Reply:
It's never useful I guess:joy: ...I did to that way because I liked to not see a stage column if there is no stage. But if we decide that it is more a problem than a nice feature we can remove it.

Reply:
Ah, yep remove it. It's too surprising and specific

Reply:
yeah, looking again now I see this all came along with the autofocus stuff in #147 ... another good example of why large PRs are tough to review.  I would go ahead and basically remove everything from #147 that wasn't strictly necessary to accomplish the collection of autofocus information.  I think that means getting rid of `_on_sys_cfg_loaded`, `_on_property_changed`, and generally all of the "core-following" stuff.

This seems to come up often: the desire to make one master widget that automagically does all the things that (e.g.) napari manager might need it to do when actively connected to a microscope session.  But I think this is probably better accomplished by subclasses for specific scenarios.  Don't do it in this PR, but after we remove this stuff, I'd like to more generally look at how deeply we've automagically connected a live core to our widgets, and consider a global concept of "follow-core".  Certainly, that's the point of many of the widgets, but it probably isn't the point of the all the widgets (like the MDA widgets) and might be better broken up into purpose-driven subclasses

Reply:
@tlambert03 I think I removed the column hiding behavior and left only the autofocus stuff the I have added in #147. And I also think that with this changes, the core is not much involved anymore, only in few places where I think it is necessary. But you can have a look and we can make fixes in another PR.